cql3/restrictions/statement_restrictions.cc

```diff
@@ -160,8 +153,8 @@ struct intersection_visitor {
     }
 
     value_set operator()(const interval<managed_bytes>& a, const value_list& b) const {
-        const auto common = b | filtered([&] (const managed_bytes& el) { return a.contains(el, type->as_tri_comparator()); });
-        return value_list(common.begin(), common.end());
+        auto common = b | std::views::filter([&] (const managed_bytes& el) { return a.contains(el, type->as_tri_comparator()); });
```

Comment:
`common` cannot be `const`. When a view is made `const`, it becomes an immutable view that cannot be modified or moved from. `std::ranges::to` requires an rvalue reference to enable move semantics and efficient range conversion. A const view prevents this optimization because:

1. Const-qualification removes the ability to move resources
2. The underlying range cannot be modified or consumed
3. Range conversion requires the ability to iterate and potentially transfer ownership of elements

Consider this example:

```cpp
// This won't compile due to const-qualification
const auto view = range | std::views::filter(...);
auto vec = view | std::ranges::to<std::vector>(); // Compilation error

// This works because it allows moving/consuming the range
auto view = range | std::views::filter(...);
auto vec = view | std::ranges::to<std::vector>(); // Compiles
```

The implementation of `std::ranges::to` relies on being able to forward the range, which requires:
- Ability to move elements
- Potential resource transfer
- Non-const iteration

A simplified conceptual implementation demonstrates this:

```cpp
template <input_range Range, typename Container>
Container to(Range&& r) {
    // Requires an rvalue reference to enable moving
    return Container(std::forward<Range>(r));
}
```

The `std::forward<Range>(r)` mechanism specifically depends on the range being non-const and movable. Const-qualification prevents this by:
- Blocking move operations
- Preventing mutation of the underlying range
- Requiring deep copying instead of efficient resource transfer

This design ensures that range conversions can be as efficient as possible, leveraging move semantics and avoiding unnecessary copying.